### PR TITLE
Enable clang-tidy on torch/csrc/api

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

### CODE REVIEW

**Review by r-barnes — APPROVED** [review]

File: torch/csrc/api/include/torch/types.h [hunk 1]
```diff
@@ -38,8 +38,8 @@ namespace torch {
 // the `func()` function defined in `at::` namespace is always hidden.
 using namespace at; // NOLINT
 
-using std::nullopt;
-using std::optional;
+using std::nullopt; // NOLINT
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/api/include/torch/types.h:41]: Can't find a way like variables or functions.

File: torch/csrc/api/src/nn/module.cpp [hunk 2]
```diff
@@ -34,6 +34,7 @@ Module::Module()
     : parameters_("Parameter"), buffers_("Buffer"), children_("Submodule") {}
 
 Module::Module(std::string name) : Module() {
+  // NOLINTNEXTLINE(cppcoreguidelines-prefer-member-initializer)
   name_ = std::move(name);
 }
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/api/src/nn/module.cpp:39]: Clang-tidy would report another warning after the change.

**Pull request closed without merging**